src/plat/pc99/machine/dev/vga/io.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2024, DornerWorks
+ *
+ * SPDX-License-Identifier: GPL-2.0-only
+ */
+#include <config.h>
+#include <arch/kernel/boot_sys.h>
+#include <arch/model/statedata.h>
+#include <machine/io.h>
+#include <plat/machine/io.h>
+#include <drivers/uart.h>
+
+#define VGA_PHYS 0xb8000 // Text buffer address
+
+//black background, light grey text, no blinking
+#define VGA_COLOR 0x07
+#define CLEAR_SCREEN_CHAR 32
+static uint64_t vga_addr = 0;
+static uint64_t vga_size = 0;
+// assume 80x25 console
+#define MAX_X_POS 80
+#define MAX_Y_POS 25
+
+/* Name kept 'serial_init' since other code expects to call this */
+void serial_init(uint16_t port)
+{
+    set_vga_addr(VGA_PHYS);
+}
+
+void set_vga_addr(uint64_t addr)
+{
+    vga_addr = addr;
+    // multiply by 2 since each slot in the VGA buffer has a char byte and 'style' byte
+    vga_size = MAX_X_POS * MAX_Y_POS * 2;
+}
+
+uint64_t get_vga_addr(void)
+{
+    return vga_addr;
+}
+
+static int verify_vga_offset(uint64_t offset)
+{
+    if (offset > vga_size) {
+        return -1;
+    }
+    return 0;
+}
+
+static void vga_write(uint16_t data, uint64_t offset)
```

Comment:
It's a bit nitpicking, but when looking at the usage, isn't this the signature to use: 
```suggestion
static void vga_write(uint16_t data, word_t x, word_t y)
```

Reply:
Could then also centralise the (x, y) to offset calculation into one function that is called for both read and write. I was thinking about suggesting all that, but left it because it'd change a lot of code without making that big a difference. It'd overall be slightly nicer to read, but the code is not that complex. On the downside it would prevent the compiler from optimising the arithmetic in the `for` loop where the scrolling happens. 

Happy to leave it up to @ericspidledw to decide, I'm fine with either version.